tests/agent_unittests/test_harvest_loop.py

```diff
@@ -555,11 +555,16 @@ def test_reservoir_sizes(transaction_node):
         "distributed_tracing.enabled": True,
     },
 )
-def test_reservoir_size_zeros(harvest_name, event_name):
+def test_reservoir_size_zeros(harvest_name, harvest_setting, event_name):
     app = Application("Python Agent Test (Harvest Loop)")
     app.connect_to_data_collector(None)
 
-    setattr(settings.event_harvest_config.harvest_limits, harvest_name, 0)
+    try:
+        setattr(getattr(settings, harvest_setting), harvest_name, 0)
+    except AttributeError:
+        # For `settings.application_logging.forwarding.max_samples_stored`
+        setattr(getattr(getattr(settings, harvest_setting.split(".")[0]), harvest_setting.split(".")[1]), harvest_name, 0)
+    
```

Comment:
And let's dynamically walk the settings object for arbitrary names, rather than this try/except logic.

```suggestion
def test_reservoir_size_zeros(harvest_setting, event_name):
    app = Application("Python Agent Test (Harvest Loop)")
    app.connect_to_data_collector(None)

    # Walk down the settings tree until the 2nd to last setting name is reached to get the
    # settings container, then set the final setting on that container to 0
    harvest_setting = list(harvest_setting.split("."))
    _settings = settings
    for setting_attr in harvest_setting[:-1]:
        _settings = getattr(_settings, setting_attr)
    setattr(_settings, harvest_setting[-1], 0)

```